homeassistant/components/rfxtrx/__init__.py

```diff
@@ -415,11 +336,18 @@ class RfxtrxEntity(RestoreEntity):
 
     def __init__(self, device, device_id, event=None):
         """Initialize the device."""
-        self._name = f"{device.type_string} {device.id_string}"
+        self._attr_unique_id = "_".join(x for x in device_id)
+        self._attr_name = f"{device.type_string} {device.id_string}"
+        self._attr_should_poll = False
+        self._attr_assumed_state = True
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, *device_id)},
+            "name": f"{device.type_string} {device.id_string}",
+            "model": device.type_string,
+        }
```

Comment:
We should not combine refactoring with a breaking change PR. Please split those.